### PR TITLE
209 golang port

### LINKED ISSUE #209: Golang fixed PORT
### Describe the bug

Golang PORT should not be fixed to a user hardcoded by the user, but Aspire should be able to randomly assign one using a custom env PORT.

### Regression

_No response_

### Steps to reproduce

```text
Currently user can only assign a PORT manually when calling `AddGolangApp()` and that needs to match the PORT to which the server is listening on.
```

### Expected behavior

User should be able to not choose a PORT value and let Aspire assign one.

### Screenshots

_No response_

### IDE and version

Other

### IDE version

_No response_

### Nuget packages

- [ ] CommunityToolkit.Aspire.Hosting.Azure.DataApiBuilder
- [ ] CommunityToolkit.Aspire.Hosting.Azure.StaticWebApps
- [ ] CommunityToolkit.Aspire.Hosting.Deno
- [x] CommunityToolkit.Aspire.Hosting.Golang
- [ ] CommunityToolkit.Aspire.Hosting.Java
- [ ] CommunityToolkit.Aspire.Hosting.Meilisearch
- [ ] CommunityToolkit.Aspire.Hosting.NodeJS.Extensions
- [ ] CommunityToolkit.Aspire.Hosting.Ollama
- [ ] CommunityToolkit.Aspire.Meilisearch
- [ ] CommunityToolkit.Aspire.OllamaSharp

### Nuget package version(s)

9.0.0-beta.54

### Additional context

_No response_

### Help us help you

Yes, I'd like to be assigned to work on this item

### PR DESCRIPTION
**Pull request by tommasodotNET**
**Closes #209 **

The port for Golang App is now configured in the AppHost. By default, it can be randomly assigned by Aspire:
```csharp
var golang = builder.AddGolangApp("golang", "../gin-api")
    .WithHttpEndpoint(env: "PORT");
```
## PR Checklist
- [x] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [x] Based off latest main branch of toolkit
- [x] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [x] New integration
  - [x] Docs are written
  - [x] Added description of major feature to project description for NuGet package (4000 total character limit, so don't push entire description over that)
- [x] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Every new API (including internal ones) has full XML docs
- [x] Code follows all style conventions

### CODE REVIEW

**Review by aaronpowell — APPROVED** [review]

**Comment by aaronpowell** [comment]
We're getting Ollama failures that should be resolved with #203

File: src/CommunityToolkit.Aspire.Hosting.Golang/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@
 Aspire.Hosting.ApplicationModel.GolangAppExecutableResource
 Aspire.Hosting.ApplicationModel.GolangAppExecutableResource.GolangAppExecutableResource(string! name, string! workingDirectory) -> void
 Aspire.Hosting.GolangAppHostingExtension
-static Aspire.Hosting.GolangAppHostingExtension.AddGolangApp(this Aspire.Hosting.IDistributedApplicationBuilder! builder, string! name, string! workingDirectory, int port = 8080, string![]? args = null) -> Aspire.Hosting.ApplicationModel.IResourceBuilder<Aspire.Hosting.ApplicationModel.GolangAppExecutableResource!>!
```
Comment (Alirexaa) [inline src/CommunityToolkit.Aspire.Hosting.Golang/PublicAPI.Shipped.txt:5]: This line should not be edited.
The following line should be added to `PublicAPI.Unshipped.txt` file instead :

```
*REMOVED*static Aspire.Hosting.GolangAppHostingExtension.AddGolangApp(this Aspire.Hosting.IDistributedApplicationBuilder! builder, string! name, string! workingDirectory, int port = 8080, string![]? args = null) -> Aspire.Hosting.ApplicationModel.IResourceBuilder<Aspire.Hosting.ApplicationModel.GolangAppExecutableResource!>!
```

Comment (aaronpowell) [reply src/CommunityToolkit.Aspire.Hosting.Golang/PublicAPI.Shipped.txt:5]: I always forget that

**Comment by Alirexaa** [comment]
@aaronpowell @tommasodotNET We should update docs, Right?

**Comment by aaronpowell** [comment]
Wait, looks like we're all good - https://learn.microsoft.com/en-gb/dotnet/aspire/community-toolkit/hosting-golang?tabs=dotnet-cli doesn't show how to set the port

**Comment by Alirexaa** [comment]
We export the port in the example project. Docs should cover this.

**Comment by tommasodotNET** [comment]
Yes, the `AddGolangApp` doesn't set up a HttpEndpoint as discussed [here](https://github.com/CommunityToolkit/Aspire/issues/209#issuecomment-2466469227). Docs should be updated. The readme already is, though. We just need to bring it to official docs?

**Comment by Alirexaa** [comment]
> Yes, the `AddGolangApp` doesn't set up a HttpEndpoint as discussed [here](https://github.com/CommunityToolkit/Aspire/issues/209#issuecomment-2466469227). Docs should be updated. The readme already is, though. We just need to bring it to official docs?

Yes, we should. Could you send PR to aspire-docs repo?

**Pull request merged**